node/src/actors/chain_manager/mining.rs

```diff
@@ -400,16 +442,28 @@ impl ChainManager {
                         .map_err(move |e| {
                             log::error!("Couldn't resolve rad request {}: {}", dr_pointer, e)
                         })
+                        .into_actor(act)
                 })
-                .flatten()
-                .and_then(move |(vrf_proof, reveal_value)| {
+                .then(|res, _, _| {
+                    // This is .flatten()
+                    match res {
+                        Ok(Ok(x)) => actix::fut::ok(x),
+                        Ok(Err(())) => actix::fut::err(()),
+                        Err(()) => actix::fut::err(()),
+                    }
```

Comment:
It is `.flatten()` but...

Reply:
But if you put `.flatten()` it doesn't compile :man_shrugging:

Reply:
Gotcha.